### PR TITLE
Indicate project resolution failure with a flag

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: minor
Connects-to: #14
Signed-off-by: Cameron Diver <cameron@resin.io>
---- Autogenerated Waffleboard Connection: Connects to #14

### CODE REVIEW

File: lib/build-task.ts [hunk 1]
```diff
@@ -83,4 +83,8 @@ export interface BuildTask {
 	 * the composition
 	 */
 	serviceName: string;
+	/**
+	 * Has this task failed to be resolved?
+	 */
+	resolveFailure: boolean;
```
Comment (dfunckt) [inline lib/build-task.ts]: I think `resolveFailure` makes me expect it's of `Error` type, not a boolean -- let's either store the `ProjectResolutionError` here (if you think that contains useful info) or rename the attribute to `resolved`.

Comment (CameronDiver) [reply lib/build-task.ts]: Yeah it's a bit weird with `BuildTask` being an interface, this need to be defined first, but you're right having it intialised with `false` makes more sense and naming it `resolved`.

File: lib/resolve.ts [hunk 2]
```diff
@@ -44,10 +43,12 @@ export function resolveTask(
 	.then((res: Resolve.ResolvedBundle) => {
 		task.projectType = res.projectType;
 		task.buildStream = res.tarStream;
+		task.resolved = true;
 		return task;
 	})
-	.catch((e: Error) => {
-		throw new ProjectResolutionError(e);
+	.catch(() => {
+		task.projectType = 'Could not be detected';
```
Comment (dfunckt) [inline lib/resolve.ts]: Doesn't need `task.resolved = false;` here as well?

Comment (CameronDiver) [reply lib/resolve.ts]: It's not strictly necessary due to the flag having to be defined from the start: https://github.com/resin-io-modules/resin-multibuild/pull/15/files#diff-37a775e32e4a1ab95c3a6ada682775dcR45 

If you think it makes things clearer though I'm happy to add it in?

**Comment by CameronDiver** [comment]
From a caller's perspective, things can continue as normal, the docker daemon will just throw an error saying that it cannot find a Dockerfile. I guess given this, it's not actually a major change...

File: lib/resolve.ts (hunk 2 above)
Comment (dfunckt) [reply lib/resolve.ts]: Ah, I see, no then that's fine.

**Comment by dfunckt** [comment]
Yeah I think a minor version bump is fine -- I don't use ProjectResolutionError in the CLI and I think we can be cheeky as I doubt there are users of this lib except the CLI and Builder at the moment.

**Comment by CameronDiver** [comment]
I did think that, I just wasn't sure :p ok yeah I would prefer only a minor bump tbh.

**Pull request merged**